tmail-backend/mailbox/encrypted/pgp-utils/src/test/java/com/linagora/tmail/pgp/ClearEmailContentExtractorTest.java

```diff
@@ -0,0 +1,221 @@
+            .setBody(ATTACHMENT_CONTENT, "plain", StandardCharsets.UTF_8)
+            .setContentDisposition("attachment");
+    }
+
+    @Test
+    void extractShouldSuccessWhenBinaryContentOnly() throws IOException {
+        Message message = Message.Builder.of()
+            .setBody(BasicBodyFactory.INSTANCE.binaryBody(BINARY_CONTENT, StandardCharsets.UTF_8))
+            .build();
+
+        assertThat(testee.extract(message))
+            .isEqualTo(ClearEmailContent.builder()
+                .html("")
+                .preview(Preview.EMPTY)
+                .attachments(ImmutableList.of()));
+    }
+
+    @Test
+    void extractShouldSuccessWhenMultipartMixed() throws IOException {
+        Multipart multipart = MultipartBuilder.create("mixed")
+            .addBodyPart(textAttachment)
+            .addBodyPart(htmlPart)
+            .addBodyPart(textPart)
+            .build();
+        Message message = Message.Builder.of()
+            .setBody(multipart)
+            .build();
+        ClearEmailContent clearEmailContent = testee.extract(message);
+
+        assertSoftly(Throwing.consumer(softly -> {
+            softly.assertThat(clearEmailContent.getHtml())
+                .isEqualTo(HTML_CONTENT);
+            softly.assertThat(clearEmailContent.getPreview())
+                .isEqualTo(previewFactory.fromMime4JMessage(message));
+            softly.assertThat(clearEmailContent.isHasAttachment())
+                .isTrue();
+            softly.assertThat(clearEmailContent.getAttachments()
+                .get(0)
+                .getContent()
+                .contentEquals(ByteSource.wrap(ATTACHMENT_CONTENT.getBytes(StandardCharsets.UTF_8))))
+                .isTrue();
+        }));
+    }
+
+    @Test
+    void attachmentShouldBeReturnEmptyWhenMessageHaveNotAttachment() throws IOException {
+        Message message = Message.Builder.of()
+            .setSubject("Subject")
+            .setBody("small message", StandardCharsets.UTF_8)
+            .build();
+        ClearEmailContent clearEmailContent = testee.extract(message);
+        assertThat(clearEmailContent.isHasAttachment())
+            .isFalse();
+        assertThat(clearEmailContent.getAttachments())
+            .isEmpty();
+    }
+
+    @Test
+    void attachmentShouldBeReturnValueWhenMessageHasSingleAttachment() throws IOException {
+        Multipart multipart = MultipartBuilder.create("mixed")
+            .addBodyPart(textAttachment)
+            .addBodyPart(htmlPart)
+            .addBodyPart(textPart)
+            .build();
+        Message message = Message.Builder.of()
+            .setBody(multipart)
+            .build();
+
+        ClearEmailContent actual = testee.extract(message);
+        assertThat(actual.isHasAttachment())
+            .isEqualTo(true);
+        assertThat(actual.getAttachments())
+            .hasSize(1);
+    }
+
+    @Test
+    void attachmentShouldBeReturnValueWhenMessageHasMultiAttachments() throws IOException {
```

Comment:
```
    void attachmentShouldReturnValueWhenMessageHasMultiAttachments() throws IOException {
```
In fact same remark for most of your test names, remove the `Be` in the middle please :)